src/common/cockpitpipe.c

```diff
@@ -273,20 +276,37 @@ dispatch_input (gint fd,
    */
   if (cond != G_IO_HUP)
     {
-      g_debug ("%s: reading input", self->priv->name);
+      g_byte_array_set_size (self->priv->in_buffer, len + MAX_SEQ_PACKET);
+      if (self->priv->seq_packet)
+        {
+          g_debug ("%s: receiving input", self->priv->name);
+          vec.iov_len = MAX_SEQ_PACKET;
+          vec.iov_base = self->priv->in_buffer->data + len;
+          ret = recvmsg (self->priv->in_fd, &msg, 0);
+        }
+      else
+        {
+          g_debug ("%s: reading input %x", self->priv->name, cond);
+          ret = read (self->priv->in_fd, self->priv->in_buffer->data + len, MAX_SEQ_PACKET);
```

Comment:
ok, i still think it reads a little weird if we are not a seqpacket pipe use the MAX_SEQ_PACKET size. But not a big deal